uaclient/cli.py

```diff
@@ -97,6 +98,38 @@ def new_f(args, cfg):
     return new_f
 
 
+def daemon_file_exists():
+    return os.path.exists(defaults.DEFAULT_UPGRADE_CONTRACT_FLAG_FILE)
+
+
+def get_running_services():
+    running_services, _ = util.subp(["ps", "-aux"])
+    return running_services
+
+
+def assert_contract_upgrade_not_running(f):
+    """Decorator asserting contract upgrade is not running."""
+
+    @wraps(f)
+    def new_f(args, cfg):
+        if daemon_file_exists():
```

Comment:
this is a good idea with the decorator and something we should walk through with rick today to get his thoughts on how best to expose this information to the user for this release.

I feel like this damon status check info is something that we don't want to just hide inside this decorator.

I believe we will ultimately also want to write into status.json too, somehting machine readable for two reasons:
 1. we might want to allow status to succeed and report another bit of info "in-progress" or something while still emitting seatus output 
 2. we might want to add a blocking ua status --wait call so that PRO machines which setup a lot of services on first boot may not want to start using those VMs until all security patches etc are applied.